query-engine/js-connectors/smoke-test-js/src/connector/planetscale.ts

```diff
@@ -174,8 +167,35 @@ class PrismaPlanetScale implements Connector, Closeable {
    * example. The version string is returned directly without any form of
    * parsing or normalization.
    */
-  version(): Promise<string | undefined> {
-    return Promise.resolve(this.maybeVersion)
+  async version(): Promise<string | undefined> {
+    if (this._version) {
+      return Promise.resolve(this._version)
+    }
+
+    const { rows } = await this.performIO({ sql: 'SELECT @@version', args: [] })
+    const version = rows[0]['@@version'] as string
+    return version
+  }
+
+  /**
+   * Run a query against the database, returning the result set.
+   * Should the query fail due to a connection error, the connection is
+   * marked as unhealthy.
+   */
+  private async performIO(query: Query) {
+    const { sql, args: values } = query
+
+    try {
+      return await this.client.execute(sql, values, { as: 'object' })
+    } catch (e) {
+      const error = e as Error & { code: string }
+      
+      if (['ENOTFOUND', 'EAI_AGAIN'].includes(error.code)) {
+        this._isHealthy = false
+      }
```

Comment:
These error codes are returned by Node.js' `fetch` when the driver can't connect to the database. Note that other, generic errors (e.g., unique constraint violations) don't result in `this._isHealthy` to be set to `false`.